drivers/watchdog/wdt_mchp_xec.c

```diff
@@ -0,0 +1,188 @@
+#include <assert.h>
+
+#include "wdt_mchp_xec.h"
+
+#define MEC_WDT_INT_SRC_VAL	BIT(2)
+
+#define WDT_XEC_REG_BASE				\
+	((volatile struct wdt_xec_registers *)		\
+	 (DT_INST_0_MICROCHIP_XEC_WATCHDOG_BASE_ADDRESS))
+
+DEVICE_DECLARE(wdt_xec);
+
+static int wdt_xec_setup(struct device *dev, u8_t options)
+{
+	volatile struct wdt_xec_registers *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (wdt_regs->control_reg.wdt_enable) {
+		return -EBUSY;
+	}
+
+	if (!data->timeout_installed) {
+		LOG_ERR("No valid WDT timeout installed");
+		return -EINVAL;
+	}
+
+	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
+		LOG_WRN("WDT_OPT_PAUSE_IN_SLEEP is not supported");
+		return -ENOTSUP;
+	}
+
+	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
+		wdt_regs->control_reg.jtag_stall = 1;
+	}
+
+	wdt_regs->control_reg.wdt_enable = 1;
+
+	LOG_DBG("WDT Setup and enabled");
+
+	return 0;
+}
+
+static int wdt_xec_disable(struct device *dev)
+{
+	volatile struct wdt_xec_registers *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (!wdt_regs->control_reg.wdt_enable) {
+		return -EALREADY;
+	}
+
+	wdt_regs->control_reg.wdt_enable = 0;
+	data->timeout_installed = false;
+
+	LOG_DBG("WDT Disabled");
+
+	return 0;
+}
+
+static int wdt_xec_install_timeout(struct device *dev,
+				   const struct wdt_timeout_cfg *config)
+{
+	volatile struct wdt_xec_registers *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+
+	if (wdt_regs->control_reg.wdt_enable) {
+		return -EBUSY;
+	}
+
+	if (config->window.min > 0U) {
+		data->timeout_installed = false;
+		return -EINVAL;
+	}
+
+	wdt_regs->load_reg = 0;
+
+	data->cb = config->callback;
+	if (data->cb) {
+		wdt_regs->control_reg.wdt_reset = 1;
+		wdt_regs->int_enable_reg.wdt_int_enable = 1;
+
+		LOG_DBG("WDT callback enabled");
+	} else {
+		/* Setting WDT_FLAG_RESET_SOC or not will have no effect:
+		 * even after the cb, if anything is done, SoC will reset
+		 */
+		wdt_regs->control_reg.wdt_reset = 0;
+		wdt_regs->int_enable_reg.wdt_int_enable = 0;
+
+		LOG_DBG("WDT Reset enabled");
+	}
+
+	/* Since it almost takes 1ms to decrement the load register
+	 * (See datasheet 18.6.1.4: 33/32.768 KHz = 1.007ms)
+	 * Let's use the given window directly.
+	 */
+	wdt_regs->load_reg = config->window.max;
+
+	data->timeout_installed = true;
+
+	return 0;
+}
+
+static int wdt_xec_feed(struct device *dev, int channel_id)
+{
+	volatile struct wdt_xec_registers *wdt_regs = WDT_XEC_REG_BASE;
+
+	ARG_UNUSED(dev);
+	ARG_UNUSED(channel_id);
+
+	if (!wdt_regs->control_reg.wdt_enable) {
+		return -EINVAL;
+	}
+
+	LOG_DBG("WDT Kicking");
+
+	wdt_regs->kick_reg = 1;
+
+	return 0;
+}
+
+static void wdt_xec_isr(struct device *dev)
+{
+	volatile struct wdt_xec_registers *wdt_regs = WDT_XEC_REG_BASE;
+	struct wdt_xec_data *data = dev->driver_data;
+	u32_t girq_result;
+
+	girq_result = MCHP_GIRQ_RESULT(MCHP_GIRQ21_ID);
+	REG32(MCHP_GIRQ_SRC_ADDR(MCHP_GIRQ21_ID)) = girq_result;
```

Comment:
You should not clear all the result bits as there are other devices  in GIRQ21. Imaging you have RTC isrs (bit9 and bit8)  turned in addition to bit 2.   If you continue with this approach you are also cleaning them and they should be clear in their respective isr handler.